components/scream/src/physics/shoc/atmosphere_macrophysics.hpp

```diff
@@ -89,23 +89,15 @@ class SHOCMacrophysics : public scream::AtmosphereProcess
       const Real ggr = C::gravit;
 
       const int nlev_packs = ekat::npack<Spack>(nlev);
-      const int num_qtracer_packs = ekat::npack<Spack>(num_qtracers);
-
-      const int nlevi_v = nlev/Spack::n;
-      const int nlevi_p = nlev%Spack::n;
-
-      const auto sub_z_int = ekat::subview(z_int, i);
-      const auto s_z_int = ekat::scalarize(sub_z_int);
       Kokkos::parallel_for(Kokkos::TeamThreadRange(team, nlev_packs), [&] (const Int& k) {
         const auto range = ekat::range<IntSmallPack>(k*Spack::n);
         const Smask in_nlev_range = (range < nlev);
 
         // Inverse of Exner. Assert that exner != 0 when in range before computing.
-        const Spack p_mid_ik(p_mid(i,k));
-        const Spack exner_ik = PF::exner_function(p_mid_ik);
-        const Smask nonzero = (exner_ik != 0);
+        const Spack exner = PF::exner_function(p_mid(i,k));
+        const Smask nonzero = (exner != 0);
         EKAT_KERNEL_ASSERT((nonzero || !in_nlev_range).any());
```

Comment:
What's the reason behind the kernel assert above? I would expect (nonzero || !in_nlev_range).all(), no?